src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    // якщо key.type === 'addProperties'
+    if (action.type === 'addProperties') {
+      // додаємо name: 'Jim' до state
+      Object.assign(obj, action.extraData);
+
+      result.push({ ...obj });
+    }
+
+    if (action.type === 'removeProperties') {
```

Comment:
Try to use switch case construction for better readability

Reply:
not sure, if it`s appropriate to use if I use switch case or I just have lack of knowledge